pyuvdata/parameter.py

```diff
@@ -1021,38 +1021,42 @@ def __init__(
             tols=(0, radian_tol),
         )
 
-    def __eq__(self, other):
+    def __eq__(self, other, silent=False):
```

Comment:
Wouldn't it be better just to use the `logging` module instead of this `silent` parameter?

Reply:
That would mean you don't have to have a parameter of `__eq__` etc, and still maintain the ability for granular output.

Reply:
That might be a good option. I'll need to look into that more. If we change away from this I'll need to make corresponding updates in pyradiosky.

Reply:
Reading here: https://docs.python.org/3/howto/logging.html#when-to-use-logging
I'm uncertain. So the existing behavior is that if you call `__eq__` is that if the objects are not equal, it prints some helpful info about why they are not equal. This can be enormously helpful. But when you call `__ne__`, you don't really need that info. The worry I have about using logging is that the standard user will just stop getting this info and will not know that they can get it by adjusting the logging level. I think if you poll our users, most would not think to adjust that.

Reply:
Yeah, I considered that, and maybe you're right. I also don't think it's hugely obvious that you should call `this.__ne__(other, silent=False)` instead of just `this != other`. Logging is fairly standard practice. 

The fact that you _sometimes_ want to print the info and sometimes not is a bit more concerning (and not naively solvable within logging). I wonder if perhaps a better route here is to have a separate function like `uvd.print_diff(other)` where a user can explicitly print such a diff?

Reply:
So the way it's set up now, if you call `this != other` you won't get any messages but if you call `this == other` you will. In the tests I called it as `this.__ne__(other, silent=False)` in order to cover the lines of code that print things, but I wouldn't think most people would need/want to call it that way.